In [28]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [29]:
df = pd.read_csv("/kaggle/input/world-happiness/2019.csv")
df.head()

Overall rank Country or region  Score  GDP per capita  Social support  \
0             1           Finland  7.769           1.340           1.587   
1             2           Denmark  7.600           1.383           1.573   
2             3            Norway  7.554           1.488           1.582   
3             4           Iceland  7.494           1.380           1.624   
4             5       Netherlands  7.488           1.396           1.522   

   Healthy life expectancy  Freedom to make life choices  Generosity  \
0                    0.986                         0.596       0.153   
1                    0.996                         0.592       0.252   
2                    1.028                         0.603       0.271   
3                    1.026                         0.591       0.354   
4                    0.999                         0.557       0.322   

   Perceptions of corruption  
0                      0.393  
1                      0.410  
2                      0.341  
3                      0.118  
4                      0.298

In [30]:
df.shape

(156, 9)

the dimension of the dataset is 156,9

In [31]:
df.head()

Overall rank Country or region  Score  GDP per capita  Social support  \
0             1           Finland  7.769           1.340           1.587   
1             2           Denmark  7.600           1.383           1.573   
2             3            Norway  7.554           1.488           1.582   
3             4           Iceland  7.494           1.380           1.624   
4             5       Netherlands  7.488           1.396           1.522   

   Healthy life expectancy  Freedom to make life choices  Generosity  \
0                    0.986                         0.596       0.153   
1                    0.996                         0.592       0.252   
2                    1.028                         0.603       0.271   
3                    1.026                         0.591       0.354   
4                    0.999                         0.557       0.322   

   Perceptions of corruption  
0                      0.393  
1                      0.410  
2                      0.341  
3                      0.118  
4                      0.298

In [32]:
df.tail(5)

Overall rank         Country or region  Score  GDP per capita  \
151           152                    Rwanda  3.334           0.359   
152           153                  Tanzania  3.231           0.476   
153           154               Afghanistan  3.203           0.350   
154           155  Central African Republic  3.083           0.026   
155           156               South Sudan  2.853           0.306   

     Social support  Healthy life expectancy  Freedom to make life choices  \
151           0.711                    0.614                         0.555   
152           0.885                    0.499                         0.417   
153           0.517                    0.361                         0.000   
154           0.000                    0.105                         0.225   
155           0.575                    0.295                         0.010   

     Generosity  Perceptions of corruption  
151       0.217                      0.411  
152       0.276                      0.147  
153       0.158                      0.025  
154       0.235                      0.035  
155       0.202                      0.091

In [33]:
df.sample(5)

Overall rank Country or region  Score  GDP per capita  Social support  \
137           138            Zambia  4.107           0.578           1.058   
45             46            Kosovo  6.100           0.882           1.232   
64             65              Peru  5.697           0.960           1.274   
94             95            Bhutan  5.082           0.813           1.321   
55             56           Jamaica  5.890           0.831           1.478   

     Healthy life expectancy  Freedom to make life choices  Generosity  \
137                    0.426                         0.431       0.247   
45                     0.758                         0.489       0.262   
64                     0.854                         0.455       0.083   
94                     0.604                         0.457       0.370   
55                     0.831                         0.490       0.107   

     Perceptions of corruption  
137                      0.087  
45                       0.006  
64                       0.027  
94                       0.167  
55                       0.028

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 156 entries, 0 to 155
Data columns (total 9 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Overall rank                  156 non-null    int64  
 1   Country or region             156 non-null    object 
 2   Score                         156 non-null    float64
 3   GDP per capita                156 non-null    float64
 4   Social support                156 non-null    float64
 5   Healthy life expectancy       156 non-null    float64
 6   Freedom to make life choices  156 non-null    float64
 7   Generosity                    156 non-null    float64
 8   Perceptions of corruption     156 non-null    float64
dtypes: float64(7), int64(1), object(1)
memory usage: 11.1+ KB


There are a 9 columns in the dataset
We can notice there are 3 data types int(1),object(1), float(7)

In [42]:
df["Country or region"].unique()

array(['Finland', 'Denmark', 'Norway', 'Iceland', 'Netherlands',
       'Switzerland', 'Sweden', 'New Zealand', 'Canada', 'Austria',
       'Australia', 'Costa Rica', 'Israel', 'Luxembourg',
       'United Kingdom', 'Ireland', 'Germany', 'Belgium', 'United States',
       'Czech Republic', 'United Arab Emirates', 'Malta', 'Mexico',
       'France', 'Taiwan', 'Chile', 'Guatemala', 'Saudi Arabia', 'Qatar',
       'Spain', 'Panama', 'Brazil', 'Uruguay', 'Singapore', 'El Salvador',
       'Italy', 'Bahrain', 'Slovakia', 'Trinidad & Tobago', 'Poland',
       'Uzbekistan', 'Lithuania', 'Colombia', 'Slovenia', 'Nicaragua',
       'Kosovo', 'Argentina', 'Romania', 'Cyprus', 'Ecuador', 'Kuwait',
       'Thailand', 'Latvia', 'South Korea', 'Estonia', 'Jamaica',
       'Mauritius', 'Japan', 'Honduras', 'Kazakhstan', 'Bolivia',
       'Hungary', 'Paraguay', 'Northern Cyprus', 'Peru', 'Portugal',
       'Pakistan', 'Russia', 'Philippines', 'Serbia', 'Moldova', 'Libya',
       'Montenegro', 'Tajikis

In [44]:
print("no.of unique countries: ",df["Country or region"].nunique())

no.of unique countries:  156


**Checking for missing values**

In [45]:
df.isnull().mean()*100

Overall rank                    0.0
Country or region               0.0
Score                           0.0
GDP per capita                  0.0
Social support                  0.0
Healthy life expectancy         0.0
Freedom to make life choices    0.0
Generosity                      0.0
Perceptions of corruption       0.0
dtype: float64

there is no null values in the dataset

**Statistical Overview**

In [46]:
df.describe().transpose()

count       mean        std    min       25%  \
Overall rank                  156.0  78.500000  45.177428  1.000  39.75000   
Score                         156.0   5.407096   1.113120  2.853   4.54450   
GDP per capita                156.0   0.905147   0.398389  0.000   0.60275   
Social support                156.0   1.208814   0.299191  0.000   1.05575   
Healthy life expectancy       156.0   0.725244   0.242124  0.000   0.54775   
Freedom to make life choices  156.0   0.392571   0.143289  0.000   0.30800   
Generosity                    156.0   0.184846   0.095254  0.000   0.10875   
Perceptions of corruption     156.0   0.110603   0.094538  0.000   0.04700   

                                  50%        75%      max  
Overall rank                  78.5000  117.25000  156.000  
Score                          5.3795    6.18450    7.769  
GDP per capita                 0.9600    1.23250    1.684  
Social support                 1.2715    1.45250    1.624  
Healthy life expectancy        0.7890    0.88175    1.141  
Freedom to make life choices   0.4170    0.50725    0.631  
Generosity                     0.1775    0.24825    0.566  
Perceptions of corruption      0.0855    0.14125    0.453

We came know about the central tendency
Mean -  Avg. of all data
Median -the value that occours most frequently
Mode - Middle value with equal observation in the left and right

In [47]:
df["Overall rank"].skew()

0.0

It is a symmetrical distrubution around the mean

**Duplicate**

In [48]:
df.duplicated().sum()

0

there is no duplicate values in the dataset


In [49]:
from sklearn.preprocessing import  LabelEncoder


In [69]:
encoder =  LabelEncoder()
df["Country or region"] = encoder.fit_transform(df["Country or region"])

In [51]:
df.columns

Index(['Overall rank', 'Country or region', 'Score', 'GDP per capita',
       'Social support', 'Healthy life expectancy',
       'Freedom to make life choices', 'Generosity',
       'Perceptions of corruption'],
      dtype='object')

**Correlation Analysis**

In [52]:
df.corr().transpose()

Overall rank  Country or region     Score  \
Overall rank                      1.000000           0.009578 -0.989096   
Country or region                 0.009578           1.000000 -0.023039   
Score                            -0.989096          -0.023039  1.000000   
GDP per capita                   -0.801947           0.021662  0.793883   
Social support                   -0.767465           0.135603  0.777058   
Healthy life expectancy          -0.787411          -0.026412  0.779883   
Freedom to make life choices     -0.546606           0.041065  0.566742   
Generosity                       -0.047993           0.114529  0.075824   
Perceptions of corruption        -0.351959           0.029526  0.385613   

                              GDP per capita  Social support  \
Overall rank                       -0.801947       -0.767465   
Country or region                   0.021662        0.135603   
Score                               0.793883        0.777058   
GDP per capita                      1.000000        0.754906   
Social support                      0.754906        1.000000   
Healthy life expectancy             0.835462        0.719009   
Freedom to make life choices        0.379079        0.447333   
Generosity                         -0.079662       -0.048126   
Perceptions of corruption           0.298920        0.181899   

                              Healthy life expectancy  \
Overall rank                                -0.787411   
Country or region                           -0.026412   
Score                                        0.779883   
GDP per capita                               0.835462   
Social support                               0.719009   
Healthy life expectancy                      1.000000   
Freedom to make life choices                 0.390395   
Generosity                                  -0.029511   
Perceptions of corruption                    0.295283   

                              Freedom to make life choices  Generosity  \
Overall rank                                     -0.546606   -0.047993   
Country or region                                 0.041065    0.114529   
Score                                             0.566742    0.075824   
GDP per capita                                    0.379079   -0.079662   
Social support                                    0.447333   -0.048126   
Healthy life expectancy                           0.390395   -0.029511   
Freedom to make life choices                      1.000000    0.269742   
Generosity                                        0.269742    1.000000   
Perceptions of corruption                         0.438843    0.326538   

                              Perceptions of corruption  
Overall rank                                  -0.351959  
Country or region                              0.029526  
Score                                          0.385613  
GDP per capita                                 0.298920  
Social support                                 0.181899  
Healthy life expectancy                        0.295283  
Freedom to make life choices                   0.438843  
Generosity                                     0.326538  
Perceptions of corruption                      1.000000

* Strong negative correlation (-0.989), indicating better scores lead to better ranks.
* Highly correlated with GDP per capita (0.794), Social support (0.777), Healthy life expectancy (0.780), and Freedom to make life choices (0.567). Wealth, support systems, and health are key drivers of higher scores.
* Strongly linked to Healthy life expectancy (0.835) and Social support (0.755), suggesting wealthier countries are healthier and better supported.

* Wealth, social support, and health are the most critical factors influencing a country's overall well-being and rank.